src/main/java/com/sprint/mission/discodeit/config/SecurityConfig.java

```diff
@@ -0,0 +1,166 @@
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.security.web.authentication.logout.CompositeLogoutHandler;
+import org.springframework.security.web.authentication.logout.CookieClearingLogoutHandler;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+import org.springframework.security.web.authentication.rememberme.JdbcTokenRepositoryImpl;
+import org.springframework.security.web.authentication.rememberme.PersistentTokenRepository;
+import org.springframework.security.web.context.HttpSessionSecurityContextRepository;
+import org.springframework.security.web.context.SecurityContextRepository;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.csrf.CsrfTokenRequestAttributeHandler;
+import org.springframework.security.web.session.HttpSessionEventPublisher;
+
+@Configuration
+@RequiredArgsConstructor
+public class SecurityConfig {
+
+  private final ObjectMapper objectMapper;
+  private final UserMapper userMapper;
+  private final CustomUserDetailsService userDetailsService;
+
+  @Bean
+  SecurityFilterChain chain(HttpSecurity http,
+      PersistentTokenRepository repository) throws Exception {
+    CsrfTokenRequestAttributeHandler requestHandler = new CsrfTokenRequestAttributeHandler();
+    requestHandler.setCsrfRequestAttributeName("_csrf");
+
+    CookieCsrfTokenRepository tokenRepository = CookieCsrfTokenRepository.withHttpOnlyFalse();
+    tokenRepository.setCookieName("CSRF-TOKEN");
+    tokenRepository.setHeaderName("X-CSRF-TOKEN");
+
+    http
+        .csrf(csrf -> csrf
+            .csrfTokenRepository(tokenRepository)
+            .csrfTokenRequestHandler(requestHandler)
+            .ignoringRequestMatchers("/api/auth/logout"))
+        .authorizeHttpRequests(auth -> auth
+            .requestMatchers("/css/**", "/images/**", "/js/**").permitAll()
+            .requestMatchers("/swagger-ui/**", "/v3/api-docs/**").permitAll()
+            .requestMatchers("/actuator/**").permitAll()
+            .requestMatchers("/api/auth/csrf-token").permitAll()
+            .requestMatchers(HttpMethod.POST, "/api/users").permitAll()
+            .requestMatchers("/api/auth/login").permitAll()
+            .requestMatchers("/api/auth/me").authenticated()
+            .requestMatchers("/api/**").hasRole(Role.USER.name())
+            .anyRequest().authenticated())
+        .securityContext(securityContext -> securityContext.securityContextRepository(
+            securityContextRepository()))
+        .httpBasic(Customizer.withDefaults())
+        .formLogin(AbstractHttpConfigurer::disable)
+        .logout(AbstractHttpConfigurer::disable)
+        .addFilterBefore(authenticationFilter(), UsernamePasswordAuthenticationFilter.class)
```

Comment:
addFilterBefore는 `UsernamePasswordAuthenticationFilter` 전에 필터를 등록한다는 의미이기에 addFilterAt 키워드로 바꾸는게 좋을 것 같아요.  

```suggestion
        .addFilterAt(authenticationFilter(), UsernamePasswordAuthenticationFilter.class)
```

물론 지금은 폼로그인을 비활성화했기 때문에 addFilterBefore, addFilterAt 모두 동일하게 동작합니다!